src/sentry/preprod/size_analysis/tasks.py

```diff
@@ -0,0 +1,306 @@
+            preprod_artifact_id__in=[artifact_id],
+        ).select_related("preprod_artifact")
+        base_size_metrics = list(base_size_metrics_qs)
+
+        if not can_compare_size_metrics(head_size_metrics, base_size_metrics):
+            logger.info(
+                "preprod.size_analysis.compare.cannot_compare_size_metrics",
+                extra={"head_artifact_id": head_artifact.id, "base_artifact_id": artifact_id},
+            )
+            continue
+
+        head_metrics_map = build_size_metrics_map(head_size_metrics)
+        base_metrics_map = build_size_metrics_map(base_size_metrics)
+
+        for key, head_metric in head_metrics_map.items():
+            matching_base_size_metric = base_metrics_map.get(key)
+            if matching_base_size_metric:
+                logger.info(
+                    "preprod.size_analysis.compare.running_comparison",
+                    extra={"base_artifact_id": artifact_id, "head_artifact_id": head_artifact.id},
+                )
+                _run_size_analysis_comparison(head_metric, matching_base_size_metric)
+            else:
+                logger.info(
+                    "preprod.size_analysis.compare.no_matching_base_size_metric",
+                    extra={"head_artifact_id": head_artifact.id, "size_metric_id": head_metric.id},
+                )
+
+
+@instrumented_task(
+    name="sentry.preprod.tasks.manual_size_analysis_comparison",
+    # TODO: Consider using a dedicated compare queue
+    queue="default",
+    silo_mode=SiloMode.REGION,
+    taskworker_config=TaskworkerConfig(
+        namespace=attachments_tasks,
+        processing_deadline_duration=30,
+    ),
+)
+def manual_size_analysis_comparison(
+    head_size_metric_id: int,
+    base_size_metric_id: int,
+):
+    try:
+        head_size_metric = PreprodArtifactSizeMetrics.objects.get(id=head_size_metric_id)
+    except PreprodArtifactSizeMetrics.DoesNotExist:
+        logger.exception(
+            "preprod.size_analysis.compare.head_artifact_not_found",
+            extra={"head_size_metric_id": head_size_metric_id},
+        )
+        return
+
+    try:
+        base_size_metric = PreprodArtifactSizeMetrics.objects.get(id=base_size_metric_id)
+    except PreprodArtifactSizeMetrics.DoesNotExist:
+        logger.exception(
+            "preprod.size_analysis.compare.base_artifact_not_found",
+            extra={"base_size_metric_id": base_size_metric_id},
+        )
+        return
+
+    _run_size_analysis_comparison(head_size_metric, base_size_metric)
+
+
+def _run_size_analysis_comparison(
+    head_size_metric: PreprodArtifactSizeMetrics,
+    base_size_metric: PreprodArtifactSizeMetrics,
+):
+    try:
+        comparison = PreprodArtifactSizeComparison.objects.get(
+            head_size_analysis=head_size_metric,
+            base_size_analysis=base_size_metric,
+        )
+
+        # Existing comparison exists or is already running,
+        if comparison.state in [
```

Comment:
note: when we reprocess a build we'll have to delete any size metrics if we aren't already